bolt/tools/merge-fdata/merge-fdata.cpp

```diff
@@ -265,7 +252,10 @@ bool isYAML(const StringRef Filename) {
 void mergeLegacyProfiles(const SmallVectorImpl<std::string> &Filenames) {
   errs() << "Using legacy profile format.\n";
   std::optional<bool> BoltedCollection;
+  std::optional<bool> NoLbr;
   std::mutex BoltedCollectionMutex;
+  std::mutex NoLbrMutex;
```

Comment:
Can we avoid the use of mutexes here by just setting bolted_collection/no_lbr per profile and checking that after we parse all profiles?